ddp-workspace/projects/ddp-dsm-ui/src/app/field-datepicker/field-datepicker.component.ts

```diff
@@ -83,43 +90,56 @@ export class FieldDatepickerComponent implements OnInit, OnChanges {
   }
 
   public check(): void {
-    this.colorDuringPatch = true;
-    if (this._dateString != null) {
-      if (this._dateString === '') {
-        this.emitDate('');
-      } else {
-        if (this._dateString !== 'N/A' && this._dateString !== 'Not Found') {
-          const tmp = Utils.parseDate(this._dateString, this.dateFormat, this.allowUnknownDay);
-          // console.log(tmp);
-          if (tmp instanceof Date) {
-            this.datePicker = tmp;
-            if (this.datePicker != null) {
-              this.emitDate(Utils.getDateFormatted(this.datePicker, Utils.DATE_STRING));
-              this.colorDuringPatch = false;
-              this.error = null;
-            } else {
-              this.error = this.dateFormat;
-            }
-          } else if (tmp === Utils.DATE_PARTIAL) {
+    if (this._dateString === '') {
+      this.emitDate('');
+    } else {
+      if (this._dateString !== 'N/A' && this._dateString !== 'Not Found') {
+        const tmp = Utils.parseDate(this._dateString, this.dateFormat, this.allowUnknownDay);
+        // console.log(tmp);
+        if (tmp instanceof Date) {
+          this.hasDateChanged = (this.datePicker !== tmp);
+          this.datePicker = tmp;
+          if (this.datePicker != null) {
+            this.emitDate(Utils.getDateFormatted(this.datePicker, Utils.DATE_STRING));
             this.error = null;
-            this.emitDate(Utils.getPartialDateFormatted(this._dateString, this.dateFormat));
-            this.colorDuringPatch = false;
-          } else if (tmp == null) {
+          } else {
             this.error = this.dateFormat;
           }
+        } else if (tmp === Utils.DATE_PARTIAL) {
+          this.error = null;
+          this.emitDate(Utils.getPartialDateFormatted(this._dateString, this.dateFormat));
+          this.hasDateChanged = false;
+        } else if (tmp == null) {
+          this.error = this.dateFormat;
         }
       }
-    } else {
-      this.colorDuringPatch = false;
-      this.error = null;
     }
   }
 
+  private async saveDate(e: Event): Promise<void>{
```

Comment:
I think the `private` access modifier is not correctly used here, as far as this function is called from the template (field-datepicker.component.html, line 33). With the JiT compilation, it might not seem a problem, however, it will become problematic when we reach AOT compilation.
IDE complaining (WebStorm):
<img width="663" alt="Screenshot 2022-10-28 at 13 44 16" src="https://user-images.githubusercontent.com/77500504/198557742-8340aba2-cc72-4027-a248-c1f1db595955.png">

Moreover, you could read the following discussion as well: [AOT issue](https://github.com/angular/angular/issues/11978)

My suggestion would be to use the `public` access modifier.